mlflow/langchain/__init__.py

```diff
@@ -661,15 +662,23 @@ def _convert_ndarray_to_list(data):
             return data
 
         if isinstance(data, pd.DataFrame):
-            return _convert_ndarray_to_list(data.to_dict(orient="records"))
+            data = data.to_dict(orient="records")
```

Comment:
In order to batchlize the input, it shouldn't return early but should continue execute batchlizing logic.